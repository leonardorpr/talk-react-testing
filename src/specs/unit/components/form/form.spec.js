import React from 'react'
import { render, fireEvent, cleanup } from 'react-testing-library'

import Form from 'components/form'

describe('[COMPONENTS] Form', () => {
  it('Should render correctly', () => {
    const { container } = render(<Form onSubmit={jest.fn()} />)
    expect(container).toMatchSnapshot()
  })

  it('Should return if onSubmit was called successfully', () => {
    const onSubmit = jest.fn()

    const { container } = render(<Form onSubmit={onSubmit} />)
    fireEvent.click(container.querySelector('button'))
    expect(onSubmit).toHaveBeenCalled()
  })

  it('Should return field value', () => {
    cleanup()

    const { getByPlaceholderText } = render(<Form onSubmit={jest.fn()} />)
    fireEvent.change(getByPlaceholderText('Task name'), { target: { value: 'Make Dinner' } })
    fireEvent.change(getByPlaceholderText('Estimated time'), { target: { value: '30 minutes' } })

    expect(getByPlaceholderText('Task name').value).toEqual('Make Dinner')
    expect(getByPlaceholderText('Estimated time').value).toEqual('30 minutes')
  })

  it('Should return the fields blank', () => {
    cleanup()
    const onSubmit = jest.fn()

    const { container, getByPlaceholderText } = render(<Form onSubmit={onSubmit} />)
    fireEvent.change(getByPlaceholderText('Task name'), { target: { value: 'Make Dinner' } })
    fireEvent.change(getByPlaceholderText('Estimated time'), { target: { value: '30 minutes' } })

    fireEvent.click(container.querySelector('button'))
    expect(getByPlaceholderText('Task name').value).toEqual('')
    expect(getByPlaceholderText('Estimated time').value).toEqual('')
  })
})
