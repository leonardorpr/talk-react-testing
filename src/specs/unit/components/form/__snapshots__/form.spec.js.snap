// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[COMPONENTS] Form Should render correctly 1`] = `
<div>
  <form
    class="sc-bdVaJa hmXIMj"
  >
    <div
      class="sc-bwzfXH MKmUa"
    >
      <input
        class="sc-htpNat cjsJFF"
        placeholder="Task name"
        type="text"
        value=""
      />
      <input
        class="sc-htpNat cjsJFF"
        placeholder="Estimated time"
        type="text"
        value=""
      />
      <button
        class="sc-bxivhb jzHhFC"
        type="submit"
      >
        Create
      </button>
    </div>
  </form>
</div>
`;
